src/test/Dashboard/Charts/ChartCard.test.tsx

```diff
@@ -0,0 +1,229 @@
+    jest.spyOn(mockController, 'attach').mockReturnValue(of(ControllerState.NO_DATA));
+
+    renderChartCard(
+      <ChartCard theme={'light'} chartKind={'CPU Load'} duration={120} period={10} span={6} dashboardId={0} />
+    );
+
+    expect(screen.getByText('CPU Load')).toBeInTheDocument();
+    expect(screen.getByText('CHART_CARD.NO_RECORDING.TITLE')).toBeInTheDocument();
+    expect(screen.getByText('CHART_CARD.NO_RECORDING.DESCRIPTION')).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: /create/i })).toBeInTheDocument();
+  });
+
+  it('navigates to recording creation with prefilled state when empty state button clicked', async () => {
+    jest.spyOn(mockController, 'attach').mockReturnValue(of(ControllerState.NO_DATA));
+
+    const { user } = renderChartCard(
+      <ChartCard theme={'light'} chartKind={'CPU Load'} duration={120} period={10} span={6} dashboardId={0} />
+    );
+
+    expect(history.location.pathname).toBe('/');
+    await user.click(screen.getByRole('button', { name: /create/i }));
+    expect(history.location.pathname).toBe('/recordings/create');
+    expect(history.location.state).toEqual({
+      duration: -1,
+      labels: [
+        {
+          key: 'origin',
+          value: 'dashboard_metrics',
+        },
+      ],
+      maxAge: 120,
+      maxSize: 100 * 1024 * 1024,
+      name: 'dashboard_metrics',
+      restartExisting: true,
+      templateName: 'Continuous',
+      templateType: 'TARGET',
+    });
+  });
+
+  it.each([
+    ['CPU Load', 120, 5],
+    ['Heap Usage', 90, 10],
+    ['Network Utilization', 60, 15],
+    ['File I/O', 30, 20],
+  ])('renders iframe', async (chartKind: string, duration: number, period: number) => {
+    jest.spyOn(mockController, 'attach').mockReturnValue(of(ControllerState.READY));
+
+    const { container } = renderChartCard(
+      <ChartCard theme={'light'} chartKind={chartKind} duration={duration} period={period} span={6} dashboardId={0} />
+    );
+
+    const iframe = container.querySelector('iframe');
+    expect(iframe).toBeTruthy();
+    const u = new URL(iframe?.src || '');
+    expect(u.host).toBe('localhost:3000');
+    expect(u.protocol).toBe('http:');
+    const params = new URLSearchParams();
+    params.set('theme', 'light');
+    params.set('panelId', String(kindToId(chartKind)));
+    params.set('to', 'now');
+    params.set('from', `now-${duration}s`);
+    params.set('refresh', `${period}s`);
+    expect(u.searchParams.toString()).toEqual(params.toString());
+  });
+});
+
+const renderChartCard = (
+  ui: React.ReactElement,
+  {
+    services = defaultServices,
+    notifications = NotificationsInstance,
+    chartContext = mockChartContext,
+    preloadState = {},
+    store = setupStore(preloadState),
+    user = userEvent.setup(),
+    ...renderOptions
+  } = {}
+) => {
+  const Wrapper = ({ children }: PropsWithChildren<unknown>) => {
+    return (
+      <ChartContext.Provider value={chartContext}>
```

Comment:
We should probably wrap this around children only to keep the order that appears in the actually app.